crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -312,8 +311,24 @@ impl Timeline {
             return Err(UnsupportedReplyItem::MISSING_EVENT_ID);
         };
 
-        let add_mentions =
-            if content.mentions.is_some() { AddMentions::Yes } else { AddMentions::No };
+        let client = self.room().client();
+
+        // [The specification](https://spec.matrix.org/v1.10/client-server-api/#user-and-room-mentions) says:
+        //
+        // > Users should not add their own Matrix ID to the `m.mentions` property as
+        // > outgoing messages cannot self-notify.
+        //
+        // OK, so let's start by assuming we always want to mention the sender of the
+        // replied event (`reply_item`)…
+        let mut mention_the_sender_of_the_replied_event = AddMentions::Yes;
+
+        // … but if `reply_item` has been written by the current user, let's toggle
+        // to `AddMentions::No`.
+        if let Some(current_user_id) = client.user_id() {
+            if current_user_id == reply_item.sender {
+                mention_the_sender_of_the_replied_event = AddMentions::No;
+            }
+        }
```

Comment:
nit: Maybe we can avoid a bit of verbosity, the `client` local variable, and the `mut` by having something like:

```suggestion
        let mention_sender = if self.room().own_user_id() == reply_item.sender { AddMentions::No } else { AddMentions::Yes };
```

Reply:
Indeed, that's better, thanks!